backend/services/database_service.py

```
@@ -177,6 +178,46 @@ def get_performance_metrics(self, metric_name: str = None, limit: int = 100) ->
+            )
+            n_metrics = (
+                session.query(PerformanceMetric)
+                .filter(PerformanceMetric.timestamp < cutoff)
+                .delete(synchronize_session=False)
+            )
+            session.commit()
+            return {
+                "security_findings_deleted": n_findings,
+                "performance_metrics_deleted": n_metrics,
+            }
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)